src/SFML/Window/DRM/DRMContext.cpp

```diff
@@ -0,0 +1,599 @@
+        display = EGL_NO_DISPLAY;
+
+        gbm_device_destroy(my_gbm_device);
+        my_gbm_device = NULL;
+
+        close(my_drm.fd);
+
+        my_drm.fd = -1;
+        my_drm.mode = 0;
+
+        memset(&my_fds,     0, sizeof(struct pollfd));
+        memset(&my_evctx,   0, sizeof(drmEventContext));
+
+        waiting_for_flip = 0;
+
+        initialized = false;
+    }
+
+    void check_init()
+    {
+        if (initialized)
+            return;
+
+        // device: Use environment variable "SFML_DRM_DEVICE" (or NULL if not set)
+        char* device_str = getenv("SFML_DRM_DEVICE");
+        if (device_str && !*device_str)
+            device_str = NULL;
+
+        // mode: Use environment variable "SFML_DRM_MODE" (or NULL if not set)
+        char* mode_str = getenv("SFML_DRM_MODE");
+
+        // refresh: Use environment variable "SFML_DRM_REFRESH" (or 0 if not set)
+        // use in combination with mode to request specific refresh rate for the mode
+        // if multiple refresh rates for same mode might be supported
+        unsigned int vrefresh = 0;
+        char* refresh_str = getenv("SFML_DRM_REFRESH");
+
+        if (refresh_str)
+            vrefresh = static_cast<unsigned int>(atoi(refresh_str));
+
+        if (init_drm(&my_drm,
+            device_str,          // device
+            mode_str,            // requested mode
+            vrefresh) < 0)     // screen refresh rate
+        {
+            sf::err() << "Error initializing drm" << std::endl;
+            return;
+        }
+
+        my_gbm_device = gbm_create_device(my_drm.fd);
+
+        std::atexit(cleanup);
+        initialized = true;
+
+        my_fds.fd = my_drm.fd;
+        my_fds.events = POLLIN;
+        my_evctx.version = 2;
+        my_evctx.page_flip_handler = page_flip_handler;
+
+    }
+
+    bool has_ext(const char* extension_list, const char* ext)
+    {
+        const char* ptr = extension_list;
+        size_t len = strlen(ext);
+
+        if (!ptr || *ptr == '\0')
+                return false;
+
+        while (true) {
+                ptr = strstr(ptr, ext);
+                if (!ptr)
+                        return false;
+
+                if (ptr[len] == ' ' || ptr[len] == '\0')
+                        return true;
+
+                ptr += len;
+        }
+    }
+
+
+    EGLDisplay getInitializedDisplay()
+    {
+        check_init();
+
+        if (display == EGL_NO_DISPLAY)
+        {
+            const char* egl_exts_client = eglQueryString(EGL_NO_DISPLAY, EGL_EXTENSIONS);
+
+            if (has_ext(egl_exts_client, "EGL_EXT_platform_base"))
+            {
+                if ((!has_ext(egl_exts_client, "EGL_MESA_platform_gbm"))
+                       && (!has_ext(egl_exts_client, "EGL_KHR_platform_gbm")))
+                {
+                    sf::err() << "Couldn't find required EGL extension: EGL_MESA_platform_gbm or EGL_KHR_platform_gbm" << std::endl;
+                    abort(); // we can't do it without this extension
```

Comment:
Is there a more graceful escape? We shouldn't really call `std::abort()` in SFML, but allow the user to react.
(also should be `std::abort()`)

Reply:
What kind of reaction do you have in mind?

Reply:
Log the error and if necessary reset some states/clean up.

See for example what happens if we can't create a context on Windows:

https://github.com/SFML/SFML/blob/c7705a8d9a79bb2c8868518c32ae54f14220793c/src/SFML/Window/Win32/WglContext.cpp#L709-L714

Reply:
I believe all these checks are redundant. None of the functions provided by this extension is used anymore. All is handled by `egl.h`
This whole block can be removed I guess.